src/js/actions/layers.js

```diff
@@ -482,6 +482,13 @@ define(function (require, exports) {
             return Promise.resolve();
         }
 
+        var upToDateDocument = this.flux.store("document").getDocument(document.id);
```

Comment:
Well, I'll make one last-ditch effort to suggest that we add a "document exists" check to the document store, but I won't stand in the way of this PR because it seems sufficient.